swift/apple/FirezoneNetworkExtension/Adapter.swift

```diff
@@ -136,7 +136,13 @@ class Adapter {
            let encoded = try? JSONEncoder().encode(resolvers),
            let jsonResolvers = String(data: encoded, encoding: .utf8)?.intoRustString() {
 
-          session?.setDns(jsonResolvers)
+          do {
+            session?.setDns(jsonResolvers)
+          } catch let error {
+            // `toString` needed to deep copy the string and avoid a possible dangling pointer
+            let msg = (error as? RustString)?.toString() ?? "Unknown error"
+            Log.error(error)
```

Comment:
```suggestion
            Log.error(AdapterError.setDnsError(msg))
```

Unfortunately you'll need to define a new error case for these